### DISCUSSION TITLE
Change test locale?

### DISCUSSION THREAD

**Discussion by daniel-a126** [redhat-developer/vscode-extension-tester#1540] [Q&A]
Is it possible to force locale for an extest run - the same as I'd do with `--locale <locale>` command line parameter?

Thank you!

**Comment by djelinek** [comment]
hello, could you please more specify which `locale` command line parameter are you looking for? I mean are you thinking about changing language (localisation)? because only connected seems to me using selenium and that is using parameter `--lang` which is mentioned here https://stackoverflow.com/a/18646033/10284560

is this what are you looking for?

**Comment by daniel-a126** [comment]
Hey Dominik,

I'm looking for a way to run my extension tests in a specific locale using vscode-extension-tester in order to write a locale-specific test.

VSCode determines the locale from system language, or command line parameter `--locale <locale>`, or 
 from `~/.vscode/argv.json` file (but it doesn't work in the testing scenarios). See also [here](https://code.visualstudio.com/docs/getstarted/locales).

Microsoft's (now deprecated) vscode-extension-vscode had to the the same back in 2019 - see [here](https://github.com/microsoft/vscode-extension-vscode/issues/136).

Thank you and sorry for not being overly specific the first time.

**Reply by djelinek** [reply]
Thank you for clarification, now it is clear.

At the moment the `--locale` is not implemented as you already found out. So there are two options:

1) we can add `--locale` to extest cli, but then there is a requirement `You must have the appropriate Language Pack installed for the language you specify with the command-line switch. If the matching Language Pack is not installed, VS Code will display English.` which we will also need to try handle automatically or user will need to take care. So it will require some time to implement.


2) (workaround) you can handle using code as it is described here - https://code.visualstudio.com/docs/getstarted/locales#_changing-the-display-language. It can be written using ExTester API as class which will be executed once before test suite, it will do
 - execute command `Configure Display Language`
 - select 'lang'
 - restart vscode and wait it is reloaded

**Reply by amishrap4** [reply]
Hi @djelinek,
While using (work around) option 2, when we restart(by clicking on the Restart button on the dialog) the VSCode, 
then it looses the connection right away, and throws the following error:
`[main 2025-02-19T06:04:46.704Z] Extension host with pid 38420 exited with code: 0, signal: unknown.`

**Logs:**
```
Restarting VSCode instance.....
[Test]: Run the analysis for project and verify the analysis results
```
>>> Trying to Open folder on [win32] platform =>  C:\vscode-test\test-data\workspace\launchVSCode\sample_c
[main 2025-02-19T06:04:46.704Z] Extension host with pid 38420 exited with code: 0, signal: unknown.
---- error  NoSuchWindowError: no such window: target window already closed
from unknown error: web view not found
  (Session info: chrome=128.0.6613.186)
    at Object.throwDecodedError (C:\vscode-test\node_modules\selenium-webdriver\lib\error.js:521:15)
    at parseHttpResponse (C:\vscode-test\node_modules\selenium-webdriver\lib\http.js:514:13)
    at Executor.execute (C:\vscode-test\node_modules\selenium-webdriver\lib\http.js:446:28)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Driver.execute (C:\vscode-test\node_modules\selenium-webdriver\lib\webdriver.js:744:17)
    at async Driver.findElements (C:\vscode-test\node_modules\selenium-webdriver\lib\webdriver.js:1072:17) {
  remoteStacktrace: '\tGetHandleVerifier [0x00007FF790409412+29090]\n' +
    '\t(No symbol) [0x00007FF79037E239]\n' +
    '\t(No symbol) [0x00007FF79023B1DA]\n' +
    '\t(No symbol) [0x00007FF79020FAF5]\n' +
    '\t(No symbol) [0x00007FF7902BE2C7]\n' +
    '\t(No symbol) [0x00007FF7902D5EB1]\n' +
    '\t(No symbol) [0x00007FF7902B6493]\n' +
    '\t(No symbol) [0x00007FF7902809D1]\n' +
    '\t(No symbol) [0x00007FF790281B31]\n' +
    '\tGetHandleVerifier [0x00007FF79072871D+3302573]\n' +
    '\tGetHandleVerifier [0x00007FF790774243+3612627]\n' +
    '\tGetHandleVerifier [0x00007FF79076A417+3572135]\n' +
    '\tGetHandleVerifier [0x00007FF7904C5EB6+801862]\n' +
    '\t(No symbol) [0x00007FF79038945F]\n' +
    '\t(No symbol) [0x00007FF790384FB4]\n' +
    '\t(No symbol) [0x00007FF790385140]\n' +
    '\t(No symbol) [0x00007FF79037461F]\n' +
    '\tBaseThreadInitThunk [0x00007FFEF8E8259D+29]\n' +
    '\tRtlUserThreadStart [0x00007FFEF980AF38+40]\n'
}`


So how can we solve this problem ?

**Reply by niteshj12** [reply]
Hi @djelinek 
When we can expect option 1 to be available i.e. adding `--locale` parameter to the extest command.

**Reply by niteshj12** [reply]
> Thank you for clarification, now it is clear.
> 
> At the moment the `--locale` is not implemented as you already found out. So there are two options:
> 
> 1. we can add `--locale` to extest cli, but then there is a requirement `You must have the appropriate Language Pack installed for the language you specify with the command-line switch. If the matching Language Pack is not installed, VS Code will display English.` which we will also need to try handle automatically or user will need to take care. So it will require some time to implement.
> 2. (workaround) you can handle using code as it is described here - https://code.visualstudio.com/docs/getstarted/locales#_changing-the-display-language. It can be written using ExTester API as class which will be executed once before test suite, it will do
> 
> * execute command `Configure Display Language`
> * select 'lang'
> * restart vscode and wait it is reloaded

Hi @djelinek any update for us?

**Reply by djelinek** [reply]
Hello, I am sorry I was away for a while a bit struggling around. Now I am back and I will start to address all the stuff.

Once again a big sorry and thank you for your patience.

as first I will convert this into an issue to allow as plan properly using ExTester planning board.

**Reply by djelinek** [reply]
https://github.com/redhat-developer/vscode-extension-tester/issues/1837
